postgres-appliance/scripts/postgres_backup.sh

```diff
@@ -10,23 +10,34 @@ function log
 log "I was called as: $0 $@"
 
 
-WALE_ENV_DIR=$1
-shift
+readonly WALE_ENV_DIR=$1
+readonly PGDATA=$2
+NUM_TO_RETAIN=$3
 
-PGDATA=$1
-shift
-
-NUM_TO_RETAIN=$1
-shift
-
-IN_RECOVERY=$(psql -tXqAc "select pg_is_in_recovery()")
+readonly IN_RECOVERY=$(psql -tXqAc "select pg_is_in_recovery()")
 [[ $IN_RECOVERY != "f" ]] && log "Cluster is in recovery, not running backup" && exit 0
 
-# leave at least 2 base backups before creating a new one
+# leave at least 2 days base backups before creating a new one
 [[ "$NUM_TO_RETAIN" -lt 2 ]] && NUM_TO_RETAIN=2
 
-# --aws-instance-profile flag is just ignored when running in GCE.
-envdir "${WALE_ENV_DIR}" wal-e --aws-instance-profile delete --confirm retain "${NUM_TO_RETAIN}"
+readonly WAL_E="envdir $WALE_ENV_DIR wal-e --aws-instance-profile"
+
+BEFORE=""
+
+readonly NOW=$(date +%s -u)
+while read name last_modified rest; do
+    last_modified=$(date +%s -ud "$last_modified")
+    if [ $(((NOW-last_modified)/86400)) -gt $NUM_TO_RETAIN ]; then
+        if [ -z "$BEFORE" ] || [ "$last_modified" -lt "$BEFORE_TIME" ]; then
```

Comment:
we should look for the earliest backup that is within NUM_TO_RETAIN days from now (to remove everything before that)

Reply:
having another look, maybe using a better name instead of `NUM_TO_RETAIN` would make sense with this change.